### PR TITLE
Switching to alpine as base image

### PR DESCRIPTION
**Pull request by kanurag94**
Addresses #292

### CODE REVIEW

File: infra/dockerfile.base [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+FROM python:3.7.3-alpine
 
 # Setup dependencies in a cacheable step
 COPY requirements.txt /requirements.txt
-RUN apt-get update && \
-		apt-get install -y --no-install-recommends gcc g++ libgomp1 && \
+
+RUN apk update && apk upgrade && \
+		echo "manylinux1_compatible = True" > /usr/local/lib/python3.7/site-packages/_manylinux.py && \
+		apk add --no-cache gcc g++ libgomp lapack-dev git libexecinfo-dev make libc6-compat && \
```
Comment (marco-c) [inline infra/dockerfile.base]: You are not removing these packages anymore. This is going to increase the size of the resulting image.

Comment (kanurag94) [reply infra/dockerfile.base]: Sorry, dropped out of my mind. Updated PR

File: infra/dockerfile.base [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 		pip install --disable-pip-version-check --no-cache-dir -r /requirements.txt && \
-		apt-get remove -y gcc g++ && \
-		apt-get autoremove -y && \
-		rm -rf /var/lib/apt/lists/*
+		apk del gcc g++ libgomp lapack-dev git libexecinfo-dev make libc6-compat && \
+		rm -rf /var/cache/apk/*
 
 COPY . /tmp/bugbug
-RUN pip install --disable-pip-version-check --no-cache-dir /tmp/bugbug
```
Comment (marco-c) [inline infra/dockerfile.base]: Why did you remove these options?

File: infra/dockerfile.commit_retrieval [hunk 3]
```diff
@@ -4,18 +4,18 @@ COPY infra/git_hyper_blame_patch /root/git_hyper_blame_patch
 
 # Mercurial need Python2 :(
 # git and git-hyper-blame (from depot_tools) are required by the annotate pipeline.
-RUN apt-get update && \
-    apt-get install -y python python-pip git curl && \
+RUN apk update && apk upgrade && \
+    apk add --no-cache gcc g++ libgomp lapack-dev git libexecinfo-dev make libc6-compat && \
```
Comment (marco-c) [inline infra/dockerfile.commit_retrieval]: Are you sure you need gcc here? It wasn't needed before

File: infra/dockerfile.commit_retrieval [hunk 4]
```diff
@@ -4,18 +4,18 @@ COPY infra/git_hyper_blame_patch /root/git_hyper_blame_patch
...
+    apk add python-dev py-pip curl && \
```
Comment (marco-c) [inline infra/dockerfile.commit_retrieval]: Why a new line to add these? You can add them together with the other packages above

Comment (marco-c) [reply infra/dockerfile.commit_retrieval]: Or at least you have to use ` --no-cache` here too

File: infra/dockerfile.commit_retrieval [hunk 5]
```diff
@@ -4,18 +4,18 @@ COPY infra/git_hyper_blame_patch /root/git_hyper_blame_patch
...
     python2 -m pip install --disable-pip-version-check --no-cache-dir mercurial==4.8 && \
     hg clone -r 0d68b0deeca2 https://hg.mozilla.org/hgcustom/version-control-tools /version-control-tools/ && \
-    rm -r /version-control-tools/.hg /version-control-tools/ansible /version-control-tools/docs /version-control-tools/testing && \
+    rm -rf /version-control-tools/.hg /version-control-tools/ansible /version-control-tools/docs /version-control-tools/testing && \
```
Comment (marco-c) [inline infra/dockerfile.commit_retrieval]: Why this change? We want to know if one of the directories doesn't exist, so we can stop removing it

File: infra/dockerfile.base (hunk 2 above)
Comment (kanurag94) [reply infra/dockerfile.base]: I mistakenly took an old branch for reference. I'll update the patch

File: infra/dockerfile.commit_retrieval (hunk 3 above)
Comment (kanurag94) [reply infra/dockerfile.commit_retrieval]: Yes. Mercurial setup requires gcc to be installed.
I'll still look if I can reduce some dependency

File: infra/dockerfile.commit_retrieval (hunk 4 above)
Comment (kanurag94) [reply infra/dockerfile.commit_retrieval]: Will do. Thanks

File: infra/dockerfile.commit_retrieval (hunk 3 above)
Comment (marco-c) [reply infra/dockerfile.commit_retrieval]: Mmmh, that's strange, how is it working without your changes?

Comment (kanurag94) [reply infra/dockerfile.commit_retrieval]: Not sure, but python package earlier was may be sufficient to compile headers required for mercurial install.

File: infra/dockerfile.commit_retrieval (hunk 5 above)
Comment (kanurag94) [reply infra/dockerfile.commit_retrieval]: This is a mistake. Rectified in new PR. Thanks for pointing out.

File: infra/dockerfile.commit_retrieval (hunk 3 above)
Comment (kanurag94) [reply infra/dockerfile.commit_retrieval]: Oh, So earlier slim image used to install gcc as a part of "apt-get update" command

`The following additional packages will be installed:
  binutils build-essential bzip2 cpp cpp-6 dbus dirmngr dpkg-dev fakeroot file
  g++ g++-6 gcc gcc-6`

While this isn't the case with alpine image

Comment (marco-c) [reply infra/dockerfile.commit_retrieval]: Can you check if you can run the `commit_retriever` script in a Docker container with this image?

Comment (kanurag94) [reply infra/dockerfile.commit_retrieval]: I got some errors installing numpy in bug-http_service. 
I'll update you as soon as I resolve them

**Comment by marco-c** [comment]
Closing as the PR is stale. Feel free to reopen if you still intend to work on this.
